crypto/digestible/test-utils/src/inspect_ast.rs

```diff
@@ -0,0 +1,518 @@
+use core::fmt;
+use mc_crypto_digestible::{DigestTranscript, Digestible, MerlinTranscript};
+use serde_json::json;
+use std::{ops::Deref, vec::Vec};
+
+// Digestible AST node types
+
+/// Represents a node in the AST
+#[derive(Clone, Eq, PartialEq)]
+pub enum ASTNode {
```

Comment:
Could give each enum variant a short doc comment. For context, by the time i got here, I had forgotten what a `Variant` is, and when I re-read the readme, noticed that None was talked about separately from the other four:

- A primitive is a "simple" type (as opposed to a compound type that has a canonical representation as bytes.
- A sequence is a variable length sequence of values of some other type. A sequence has a length known at runtime.
- An aggregate is a fixed-length sequence of values ("fields"), of different types. Each field has a field name.
- A variant is a single value which may be one of several different types. Each possibility has an associated name, in the context of this variant. In the sequel we call this the "variant possibility name".